async/xen_api_async_unix.ml

```diff
@@ -32,18 +32,16 @@ module IO = struct
   let iter fn x = Deferred.List.iter x ~f:fn
 
   let read_line (_, ic) =
-    Reader.read_line ic >>=
-    function
-    |`Ok s -> return (Some s)
-    |`Eof -> return None
+    Reader.read_line ic
+    >>| function
+    | `Ok s -> Some s
+    | `Eof -> None
 
-  let read =
-    let buf = String.create 4096 in
-    fun (_, ic) len ->
-      Reader.read ic ~len buf >>=
-      function
-      |`Ok len' -> return (String.sub buf 0 len')
-      |`Eof -> return ""
+  let read (_, ic) len =
+    let buf = Bytes.create len in
```

Comment:
So the motivation for this change is to ensure that now different callers of the `read` function will not share the same buffer `buf`, but a new buffer will be created for each invocation?